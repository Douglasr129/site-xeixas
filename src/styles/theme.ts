/* eslint-disable quotes */
export const theme = {
	colors: {
		primaryColor: '#0a1128',
		secondaryColor: '#dc143c',
		white: '#FFFFFF',
		mediumGray: '#DDDDDD',
		darkColor: '#2d2d2d',
	},
	font: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	media: {
		lteMedium: '(max-width: 768px)',
	},
	spacing: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
