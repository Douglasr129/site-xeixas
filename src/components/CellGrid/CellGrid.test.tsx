import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { CellGrid } from '.';

describe('<CellGrid />', () => {
	it('should render a title', () => {
		renderTheme(<CellGrid title={'Teste'} />);
		const h1 = screen.getByRole('heading', { name: 'Teste' });
		expect(h1).toBeInTheDocument();
	});
	it('should render a title with margin-bottom undefined', () => {
		renderTheme(<CellGrid title={'Teste'} />);
		expect(screen.getByRole('heading', { name: 'Teste' })).toHaveStyleRule({
			margin: '3.2rem',
		});
	});
	it('should render a subtitle', () => {
		renderTheme(<CellGrid title={'Teste'} subtitle={'SupTeste'} />);
		expect(
			screen.getByRole('heading', { name: 'SupTeste' }),
		).toBeInTheDocument();
	});
	it('should render a title with margin-bottom 0', () => {
		renderTheme(<CellGrid title={'Teste'} subtitle={'SupTeste'} />);
		expect(screen.getByRole('heading', { name: 'Teste' })).toHaveStyleRule({
			'margin-bottom': '0',
		});
		expect(screen.getByRole('heading', { name: 'SupTeste' })).toHaveStyleRule({
			'margin-top': '0',
			'margin-left': '5rem',
		});
	});
	it('should render a text', () => {
		const html = '<h4>Test<h4/>';
		renderTheme(<CellGrid title={'Teste'} subtitle={'SupTeste'} text={html} />);
		expect(screen.getByRole('heading', { name: 'Test' })).toBeInTheDocument();
	});
	it('should rende elemnets HTML', () => {
		const { container } = renderTheme(
			<CellGrid title={'Teste'} subtitle={'SupTeste'} text="Text exemple" />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
