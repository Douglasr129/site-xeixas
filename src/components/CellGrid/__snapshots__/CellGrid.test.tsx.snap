// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellGrid /> should rende elemnets HTML 1`] = `
.c0 h2 {
  margin-bottom: 0;
}

.c0>h3 {
  margin-top: 0;
  margin-left: 5rem;
}

.c1 {
  color: #161421;
  font-size: 3.2rem;
  text-transform: none;
}

.c2 {
  color: #161421;
  font-size: 1.6rem;
  text-transform: none;
}

.c3 {
  color: #161421;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Teste
  </h2>
  <h3
    class="c2"
  >
    SupTeste
  </h3>
  <div
    class="c3"
  >
    Text exemple
  </div>
</div>
`;
